src/cljs/rems/administration/create_catalogue_item.cljs

```diff
@@ -241,12 +241,10 @@
 
 (defn create-catalogue-item-page []
   (let [loading? (rf/subscribe [::loading?])
-        state (r/atom nil)
-        on-pending #(reset! state {:status :pending})
-        on-success #(reset! state {:status :saved})
-        on-error #(reset! state {:status :failed :error %})
-        on-modal-close #(do (reset! state nil)
-                            (dispatch! "#/administration/catalogue-items"))]
+        {:keys [on-pending on-success on-error state-atom] :as modal-opts}
```

Comment:
I think this piece of code is unclear and surprising so it should be refactored.

Reply:
What is unclear and surprising?

Reply:
Let's talk about it tomorrow in person. I just think that it's not a structure or solution that is a common pattern so there must be a clearer solution out there.